runners/google-cloud-dataflow-java/examples-streaming/build.gradle

```diff
@@ -47,6 +47,9 @@ task windmillPreCommit(type: Test) {
      "--tempRoot=${gcsTempRoot}",
      "--runner=TestDataflowRunner",
      "--dataflowWorkerJar=${dataflowWorkerJar}",
+     // Provide job with a customizable worker jar.
+     // With legacy worker jar, containerImage is set to empty (i.e. to use the internal build).
+     // More context and discussions can be found in PR#6694.
      "--workerHarnessContainerImage=",
```

Comment:
By setting `workerHarnessContainerImage` to an empty string, we instruct to pick up the non-versioned image as explained in https://github.com/apache/beam/pull/6694. It looks like this image ignores the java version, and runs with java 8.

CI steps `Java_Examples_Dataflow_Java11` and `Java_Examples_Dataflow_Java17` are failing because the test lib target respectively java 11 and java 17.

Someone should ensure that the test image runs with proper java version